Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -25,24 +26,33 @@ import struct TSCUtility.Version
 public final class RegistryClient: Cancellable {
     private static let apiVersion: APIVersion = .v1
     private static let availabilityCacheTTL: DispatchTimeInterval = .seconds(5 * 60)
+    private static let metadataCacheTTL: DispatchTimeInterval = .seconds(60 * 60)
 
     private let configuration: RegistryConfiguration
     private let archiverProvider: (FileSystem) -> Archiver
     private let httpClient: LegacyHTTPClient
     private let authorizationProvider: LegacyHTTPClientConfiguration.AuthorizationProvider?
     private let jsonDecoder: JSONDecoder
 
-    private var checksumTOFU: PackageVersionChecksumTOFU!
+    private(set) var checksumTOFU: PackageVersionChecksumTOFU!
+    private(set) var signatureValidation: SignatureValidation!
 
     private let availabilityCache = ThreadSafeKeyValueStore<
         URL,
         (status: Result<AvailabilityStatus, Error>, expires: DispatchTime)
     >()
 
+    private let metadataCache = ThreadSafeKeyValueStore<
```

Comment:
Just an in-memory cache. I wonder if we should use sqlite-backed cache.

Reply:
I think this is a good start, we can improve later if we see it is a bottle neck